packages/odyssey-storybook/test/testcafe/spec/button.e2e.ts

```diff
@@ -0,0 +1,109 @@
+/*!
+ * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import '../framework/pageobjects/buttonPageObject'
+import {buttonBuilder, getButtonSelector} from "../framework/pageobjects/buttonPageObject";
+
+const buttonVariants = [
+  { variant: 'primary'},
+  { variant: 'secondary'},
+  { variant: 'danger'},
+  { variant: 'floating'}
+];
+
+fixture('Testing disabled button');
+
+buttonVariants.forEach( c => {
+  test(
+    'when variant is set to '.concat(c.variant),
+    async browser => {
+      const button = buttonBuilder()
+        .setType('primary')
+        .setIsDisabled(true)
+        .setVariant(c.variant)
+        .build()
```

Comment:
I think I see what's going on here. This is targeting Storybook's `iframe` and modifying the URL to turn on and off props. I'd have to think about that one.

In the past, I just made more stories for the cases we wanted to test. That's also how I did it for OneForm. I created stories specifically for testing sets of functionality.

My issue is that all of these setters are foreign and don't match React JSX (the way people would use the component). At that point, there's no reason for Storybook. We're basically doing unit tests.

I would only recommend using TestCafe for combining multiple components together for specific test case stories.

## Another option
We could also render JSX to a blank page instead if we want to unit test using TestCafe. I also prefer using UI testers over Jest. This would be a great way of UI testing individual components.

Thing is, the logistics are all wrong. If we did this, there are a few ways of doing it; none of them that good. It'd be the same as creating stories in Storybook, but each component would render to a page all by itself.

---

I'd like to discuss this further and see what we can do. I'm not satisfied with the current approach. It's difficult to maintain and requires a ton of boilerplate. It's also using a non-standard custom-built syntax which can easily be messed up by human error.

It's a lot for devs to maintain because it tightly couples your stories with your TestCafe tests.